crates/ide/src/doc_links.rs

```diff
@@ -384,14 +385,18 @@ fn map_links<'e>(
             let (link_type, link_target_s, link_name) =
                 callback(&end_link_target.take().unwrap(), &s);
             end_link_target = Some(CowStr::Boxed(link_target_s.into()));
-            end_link_type = link_type;
+            if !matches!(end_link_type, Some(link) if link == LinkType::Autolink) {
```

Comment:
```suggestion
            if !matches!(end_link_type, Some(LinkType::Autolink)) {
```
LinkType is be `StructuralPartialEq` so this should work I believe, same for the check below